languages/python/django-oso/django_oso/partial.py

```diff
@@ -91,28 +92,37 @@ def and_expr(expr: Expression, model: Model, **kwargs):
 
 
 def compare_expr(expr: Expression, _model: Model, path=(), **kwargs):
-    q = Q()
     (left, right) = expr.args
     left_path = dot_path(left)
     if left_path:
-        return COMPARISONS[expr.operator](q, "__".join(path + left_path), right)
+        return COMPARISONS[expr.operator]("__".join(path + left_path), right)
     else:
         if isinstance(right, Model):
             right = right.pk
         else:
             raise UnsupportedError(f"Unsupported comparison: {expr}")
-        return COMPARISONS[expr.operator](q, "__".join(path + ("pk",)), right)
+        return COMPARISONS[expr.operator]("__".join(path + ("pk",)), right)
 
 
 def in_expr(expr: Expression, model: Model, path=(), **kwargs):
     assert expr.operator == "In"
-    q = Q()
     (left, right) = expr.args
     right_path = dot_path(right)
     assert right_path, "RHS of in must be a dot lookup"
     right_path = path + right_path
 
     if isinstance(left, Expression):
-        return translate_expr(left, model, path=right_path, **kwargs)
+        if left.operator == "And" and not left.args:
```

Comment:
Could we add a comment about this case? (Why it's needed - since it's nuanced)

Reply:
2989407